### PR TITLE
feat: add initialRouteName property to config

### PR DESCRIPTION
**Pull request by WoLewicki**
Added `initialRouteName` property to config. When present, the screen of its name is always added to the state object on the proper level of nesting. There is no support for `initialRouteName` inside another `intialRouteName`, so if `initialRouteName` points to a navigator, it should be resolved properly by navigator itself.

### CODE REVIEW

File: packages/core/src/getStateFromPath.tsx [hunk 1]
```diff
@@ -207,13 +261,28 @@ function createNormalizedConfigs(
     // if an object is specified as the value (e.g. Foo: { ... }),
     // it has `path` property and
     // it could have `screens` prop which has nested configs
-    configs.push(createConfigItem(routeNames, value.path, value.parse));
+    if (value.initialRouteName) {
+      if (!value.screens?.[value.initialRouteName]) {
+        throw Error(
+          `No such route in ${value}'s config: ${value.initialRouteName}`
```
Comment (satya164) [inline packages/core/src/getStateFromPath.tsx]: I think this is not needed. Initial route name shouldn't need an explicit path mapping

File: packages/core/src/__tests__/getStateFromPath.test.tsx [hunk 2]
```diff
@@ -144,7 +144,6 @@ it('converts path string to initial state with config with nested screens', () =
   const path = '/foe/bar/sweet/apple/baz/jane?count=10&answer=42&valid=true';
   const config = {
     Foo: {
-      path: 'foo',
```
Comment (satya164) [inline packages/core/src/__tests__/getStateFromPath.test.tsx]: Do we need to change existing tests? This should be an additional feature, right?

Comment (WoLewicki) [reply packages/core/src/__tests__/getStateFromPath.test.tsx]: Yeah, I just did it for clarity that it is a navigator and there should be no need to navigate to it explicitly, but it doesn't break anything.

File: packages/core/src/getStateFromPath.tsx (hunk 1 above)
Comment (WoLewicki) [reply packages/core/src/getStateFromPath.tsx]: You are right, I will change it.

File: packages/core/src/getStateFromPath.tsx [hunk 3]
```diff
@@ -110,47 +119,43 @@ export default function getStateFromPath(
...
-            },
-          };
-          helper = helper.routes[0].state as InitialState;
+    let state: InitialState = { routes: [] };
+    let routeName = routeNames.shift() as string;
+    let initialRoute = findInitialRoute(routeName, initialRoutes);
+
+    parseRoute(state, initialRoute, routeName, routeNames.length === 0, params);
+
+    if (routeName.length > 0) {
```
Comment (osdnk) [inline packages/core/src/getStateFromPath.tsx]: ??
```suggestion
    if (routeNames.length > 0) {
```

File: packages/core/src/getStateFromPath.tsx [hunk 4]
```diff
@@ -110,47 +119,43 @@ export default function getStateFromPath(
...
+        initialRoute = findInitialRoute(routeName, initialRoutes);
+        parseRoute(
+          helper,
+          initialRoute,
+          routeName,
+          routeNames.length === 0,
+          params
+        );
+        if (routeNames.length > 0) {
+          helper = helper.index
```
Comment (osdnk) [inline packages/core/src/getStateFromPath.tsx]:
```suggestion
          helper = helper.routes[helper.index || 0].state as InitialState
```

Comment (osdnk) [reply packages/core/src/getStateFromPath.tsx]: and pls use better name

File: packages/core/src/getStateFromPath.tsx [hunk 5]
```diff
@@ -254,3 +272,56 @@ function findParseConfigForRoute(
...
+}
+
+function parseRoute(
+  state: InitialState,
+  initialRoute: string | undefined,
+  routeName: string,
+  isEmpty: boolean,
+  params?: Record<string, any> | undefined
+) {
+  if (isEmpty) {
```
Comment (osdnk) [inline packages/core/src/getStateFromPath.tsx:300]: looks good, but again, make it please more descriptive. Also, could you try to omit to make side effects and use `return`?

Comment (WoLewicki) [reply packages/core/src/getStateFromPath.tsx:300]: can do

**Review by satya164 — APPROVED** [review]

**Pull request merged**